skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsgrad.java

```diff
@@ -0,0 +1,89 @@
+ */
+
+package se.inera.intyg.common.ag114.v1.model.converter.certificate.question;
+
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_HEADER_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_SVAR_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_SVAR_JSON_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_SVAR_TEXT_ID;
+import static se.inera.intyg.common.ag114.v1.model.converter.RespConstants.SJUKSKRIVNINGSGRAD_UNIT_OF_MEASURE;
+import static se.inera.intyg.common.support.facade.util.ValidationExpressionToolkit.singleExpression;
+import static se.inera.intyg.common.support.facade.util.ValueToolkit.integerValue;
+
+import se.inera.intyg.common.services.texts.CertificateTextProvider;
+import se.inera.intyg.common.support.facade.model.Certificate;
+import se.inera.intyg.common.support.facade.model.CertificateDataElement;
+import se.inera.intyg.common.support.facade.model.config.CertificateDataConfigInteger;
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidation;
+import se.inera.intyg.common.support.facade.model.validation.CertificateDataValidationMandatory;
+import se.inera.intyg.common.support.facade.model.value.CertificateDataValueInteger;
+
+public class QuestionSjukskrivningsgrad {
+
+    private static final Integer MIN = 0;
+    private static final Integer MAX = 100;
+
+    public static CertificateDataElement toCertificate(String sjukskrivningsgrad, int index, CertificateTextProvider textProvider) {
+        return CertificateDataElement.builder()
+            .id(SJUKSKRIVNINGSGRAD_SVAR_ID)
+            .parent(SJUKSKRIVNINGSGRAD_HEADER_ID)
+            .index(index)
+            .config(
+                CertificateDataConfigInteger.builder()
+                    .id(SJUKSKRIVNINGSGRAD_SVAR_JSON_ID)
+                    .text(textProvider.get(SJUKSKRIVNINGSGRAD_SVAR_TEXT_ID))
+                    .unit(SJUKSKRIVNINGSGRAD_UNIT_OF_MEASURE)
+                    .min(MIN)
+                    .max(MAX)
+                    .build()
+            )
+            .value(
+                CertificateDataValueInteger.builder()
+                    .id(SJUKSKRIVNINGSGRAD_SVAR_JSON_ID)
+                    .value(isNumeric(sjukskrivningsgrad) ? Integer.valueOf(sjukskrivningsgrad) : null)
+                    .build()
+            )
+            .validation(
+                new CertificateDataValidation[]{
+                    CertificateDataValidationMandatory.builder()
+                        .questionId(SJUKSKRIVNINGSGRAD_SVAR_ID)
+                        .expression(singleExpression(SJUKSKRIVNINGSGRAD_SVAR_JSON_ID))
+                        .build()
+                }
+            )
+            .build();
+    }
+
+    private static boolean isNumeric(String stringNumber) {
```

Comment:
Maybe this function can be moved to a more general file. I saw that you added some functions to ValueToolkit below.

Reply:
Great point, moved it to ValueToolkit as suggested.